ngraph/frontend/paddlepaddle/src/place.cpp

```diff
@@ -25,17 +25,187 @@ bool PlacePDPD::is_output() const
+{
+    return m_input_ports;
+}
+
+std::shared_ptr<OutPortPlacePDPD> OpPlacePDPD::get_output_port_pdpd(const std::string& name,
+                                                                    int idx) const
+{
+    return m_output_ports.at(name)[idx];
+}
+
+std::shared_ptr<InPortPlacePDPD> OpPlacePDPD::get_input_port_pdpd(const std::string& name,
+                                                                  int idx) const
+{
+    return m_input_ports.at(name)[idx];
+}
+
+const std::shared_ptr<paddle::framework::proto::OpDesc>& OpPlacePDPD::get_desc() const
+{
+    return m_op_desc;
+}
+
+void OpPlacePDPD::add_out_port(const std::shared_ptr<OutPortPlacePDPD>& output,
+                               const std::string& name)
+{
+    m_output_ports[name].push_back(output);
+}
+
+void OpPlacePDPD::add_in_port(const std::shared_ptr<InPortPlacePDPD>& input,
+                              const std::string& name)
+{
+    m_input_ports[name].push_back(input);
+}
+
+Place::Ptr OpPlacePDPD::get_output_port(const std::string& name) const
+{
+    FRONT_END_GENERAL_CHECK(
+        m_output_ports.at(name).size() == 1,
+        "There are multiple OutputPort Places with the same name, please specify index.");
+    return m_output_ports.at(name)[0];
+}
+
+Place::Ptr OpPlacePDPD::get_input_port(const std::string& name) const
+{
+    FRONT_END_GENERAL_CHECK(
+        m_input_ports.at(name).size() == 1,
+        "There are multiple InputPort Places with the same name, please specify index.");
+    return m_input_ports.at(name)[0];
+}
+
+Place::Ptr OpPlacePDPD::get_output_port() const
+{
+    FRONT_END_GENERAL_CHECK(m_output_ports.size() == 1 &&
+                                m_output_ports.begin()->second.size() == 1,
+                            "There are multiple OutputPort Places, please specify index.");
+    return m_output_ports.begin()->second[0];
+}
+
+Place::Ptr OpPlacePDPD::get_input_port() const
+{
+    FRONT_END_GENERAL_CHECK(m_input_ports.size() == 1 && m_input_ports.begin()->second.size() == 1,
+                            "There are multiple InputPort Places, please specify name and index.");
+    return m_input_ports.begin()->second[0];
+}
+
+std::vector<Place::Ptr> OpPlacePDPD::get_consuming_operations() const
+{
+    std::vector<Place::Ptr> consuming_ops;
+    for (const auto& out_port : m_output_ports)
+    {
+        for (const auto& out_port_place : out_port.second)
+        {
+            auto consuming_ops_out = out_port_place->get_consuming_operations();
+            consuming_ops.insert(
+                consuming_ops.end(), consuming_ops_out.begin(), consuming_ops_out.end());
+        }
+    }
+    return consuming_ops;
+}
+
+std::vector<Place::Ptr> OpPlacePDPD::get_consuming_operations(const std::string& outputPortName,
+                                                              int outputPortIndex) const
+{
+    return get_output_port(outputPortName, outputPortIndex)->get_consuming_operations();
```

Comment:
Implementation looks quite generic. If it is going to be the same for other frontends - can we consider to move this to base class implementation? Same is for other similar methods